pkg/ssh/ssh_keys.go

```diff
@@ -0,0 +1,112 @@
+package ssh
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/cli/cli/v2/internal/config"
+	"github.com/cli/cli/v2/internal/run"
+	"github.com/cli/safeexec"
+)
+
+type SshContext struct {
+	ConfigDir string
```

Comment:
It feels a bit wrong to have these be public, but they need to be settable for tests. Would making it a method like `SshContext.SetConfigDir(path)` be better?

Reply:
na Go style doesnt use setters. Sometimes this means that the type is in the wrong package (when you feel like it should be unexported) and a larger refactor could help. Other times theres not much you can do about it. Seems like this is the latter case atleast for the time being